README.md

```diff
@@ -243,22 +245,42 @@ If using make with pip, there are corresponding make targets for running linting
 
 NOTE: The lint script runs a number of separate linters to simplify the development workflow, but each individual linter can be run separately if you find this useful.
 
-#### GitLint
+#### Commit Style
+
+We aim to follow a standard commit style to keep the `git log` consistent and easy to read.
+
+Much like working with code, it's great to refer to the git log, for the style we're actively using.
+
+Our overall style for commit structure and messages broadly follows the general [Zulip version control guidelines](http://zulip.readthedocs.io/en/latest/contributing/version-control.html), so we recommend reading that first.
+
+Our commit titles have slight variations from the general Zulip style, with each:
+* starting with one or more areas in lower case, followed by a colon and space
+* area being slash-separated modified files without extensions, or the type of the change
```

Comment:
Should we recommend filenames (and areas) in an alphabetically sorted fashion, so that the git log is more consistent?

Reply:
I know that some contributors do this, but there are various rules we could use for ordering, and I think it's best to not impose even more rules on contributors in this way? I'm not sure what we gain in alphabetical ordering that isn't achievable with eg. some kind of `grep`. We could sort by "size of change in file contents" or others - and they all impose various extra work which can be determined otherwise.

Instead we could explicitly mention that the order doesn't matter?

(listing files already impose extra burden, but help summarize what has changed quickly in the log titles - and too many entries sometimes indicates if a commit is too complicated - if it's not a refactor, for example)

Reply:
Right, that does add a bit of resistance; seems reasonable. :+1: We could mention that the order doesn't matter if needed later.

Reply:
as per PM I've slightly hinted that order isn't important in the first example commit title :)